hack/images

```diff
@@ -88,5 +91,10 @@ for tagName in $tagNames; do
   tagFlags="$tagFlags--tag=$tagName "
 done
 
-buildxCmd build $platformFlag $targetFlag $importCacheFlags $exportCacheFlags $tagFlags $outputFlag $(buildAttestFlags) \
+nocacheFilterFlags=""
+if [ "$GITHUB_ACTIONS" = "true" ]; then
+  nocacheFilterFlags="--no-cache-filter=buildkit-export"
```

Comment:
Should it use no cache for any of the stages? For example the build-base stage installs tools used during build; outdated versions of those won't end up in the exported image but _are_ used as part of the build process; https://github.com/moby/buildkit/blob/fd0c25c7478b0ac91206deecd24e646b2a2b3e56/Dockerfile#L39

Reply:
We would not have any cache on CI then which makes me sad but I think we could disable cache only on release. WDYT?

Reply:
Yes, would make sense to fully disable on release.

Do we have a way to prune cache? (e.g. run a cron to purge the build-cache, so that we know that regular CI (on pull requests) also uses an up-to-date cache?)

Reply:
> Do we have a way to prune cache? (e.g. run a cron to purge the build-cache, so that we know that regular CI (on pull requests) also uses an up-to-date cache?)

We don't but I think GitHub has some tooling for. https://github.com/actions/gh-actions-cache can only delete cache with a specific key but I guess we could just list keys for the main branch (`gh actions-cache list -B main`) and remove them with `gh actions-cache delete ...`.